ChiTech/ChiPhysics/PhysicsMaterial/transportxsections/prop_transportxs_00_general.cc

```diff
@@ -129,71 +133,52 @@ void chi_physics::TransportCrossSections::
 
     // Increment combo factor totals
     N_total += combo.second;
-    if (xs->is_fissile) {
+    if (xs->is_fissile)
+    {
       this->is_fissile = true;
       Nf_total += combo.second;
+
+      if (xs->num_precursors > 0)
+        Np_total += combo.second;
     }
 
-    //============================ Check number of groups
+    //============================ Define and check number of groups
     if (cross_secs.size() == 1)
-    {
       num_grps_G = xs->num_groups;
-      if (xs->is_fissile)
-        num_precursors_J = xs->num_precursors;
+    else if (xs->num_groups != num_grps_G)
+    {
+      chi_log.Log(LOG_ALLERROR)
+        << "In call to " << __FUNCTION__ << ": "
+        << "all cross-sections must have the same number of groups.";
+      exit(EXIT_FAILURE);
     }
-    else
+
+    //============================ Increment number of precursors
+    if (not xs->is_fissile and xs->num_precursors > 0)
     {
-      if (xs->num_groups != num_grps_G)
-      {
-        chi_log.Log(LOG_ALLERROR)
-          << "In call to TransportCrossSections::MakeCombined: "
-          << "all cross-sections must have the same number of groups.";
-        exit(EXIT_FAILURE);
-      }
-      if (xs->is_fissile)
-      {
-        if (num_precursors_J == 0) 
-          num_precursors_J = xs->num_precursors;
-        else
-        {
-          if (xs->num_precursors != num_precursors_J)
-          {
-            chi_log.Log(LOG_ALLERROR)
-              << "In call to TransportCrossSections::MakeCombined: "
-              << "all fissile cross-sections must have the same number "
-              << "of precursors.";
-            exit(EXIT_FAILURE);
-          }
-        }
-      }
+      chi_log.Log(LOG_ALLERROR)
+          << "In call to " << __FUNCTION__ << ": "
+          << "only fissile materials are allowed to have delayed "
+          << "neutron precursors.";
+      exit(EXIT_FAILURE);
     }
-    ++count;
-  }//for auto
+    chi_log.Log(LOG_0) << "NUMBER OF PRECURSORS" << xs->num_precursors;
+    num_precursors_J += xs->num_precursors;
+  }//for cross-section
+
+  //this should be unnecessary with the added condition check later
   if (Nf_total < 1.0e-28)
```

Comment:
the 1e-28 makes me feel there is a cm^3 units somewhere rather than a more common cm-barn. Can anyone comment on this? Are there assumed units in the code? @Naktakala @dominiccaron @wdarylhawkins

Reply:
This one is actually difficult. There are no units involved here. Please help us out here by suggesting a better minimum value that prevents the trigger that makes a cross section fissile.

Reply:
To be dimensionless, make the comparison relative to ```N_total```, for example: ```if (Nf_total <= eps * N_total) ...```, with ```eps``` on the order of machine precision or similar. Also, it is necessary to modify appropriately the fictitious value assigned to ```Nf_total``` to ensure that it is not on the same order of magnitude as any of the constituents.

Reply:
Because this PR is based upon #160, this comment is probably best addressed there. I added a check for `is_fissile` and `num_precursors > 0` in the `MakeCombined` routine before dividing by `Nf_total` or `Np_total`. This addressed the concern that initially led to the check for `Nf_total < 1.0e-28`, but does assume that the user supplies non-zero scaling factors for the cross-sections. I would propose something along the lines of this:
```suggestion  
  if (Nf_total < eps * N_total)
    this->is_fissile = false;
  if (Np_total < eps * N_total)
    this->num_precursors = 0;
```
and then later only dividing by `Nf_total` or `Np_total` if `is_fissile == true` or `num_precursors > 0`, respectively.

Reply:
Because this PR is based upon #160, this comment is probably best addressed there. I added a check for `is_fissile` and `num_precursors > 0` in the `MakeCombined` routine before dividing by `Nf_total` or `Np_total`. This addressed the concern that initially led to the check for `Nf_total < 1.0e-28`, but does assume that the user supplies non-zero scaling factors for the cross-sections. I would propose something along the lines of what is below and then later only dividing by `Nf_total` or `Np_total` if `is_fissile == true` or `num_precursors > 0`, respectively.
```suggestion  
  double eps = std::numeric_limits<double>::epsilon(); //machine precision
  if (Nf_total < eps * N_total)
    this->is_fissile = false;
  if (Np_total < eps * N_total)
    this->num_precursors = 0;
```

Reply:
This change has been pushed to #160.